include/COSMO_SAC/util.hpp

```diff
@@ -72,16 +72,18 @@ static std::vector<std::string> str_split(const std::string &s,
 }
 
 /// The following code for the trim functions was taken from http://stackoverflow.com/questions/216823/whats-the-best-way-to-trim-stdstring
-    // trim from start
+// trim from start
 static std::string& strlstrip(std::string& s) {
-    s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char c) { return !std::isspace(c); }));
     return s;
 }
+
 // trim from end
 static std::string& strrstrip(std::string& s) {
-    s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+    s.erase(std::find_if(s.rbegin(), s.rend(), [](unsigned char c) { return !std::isspace(c); }).base(), s.end());
```

Comment:
Replaced deprecated `std::not1` and `std::ptr_fun` with lambda functions in `strlstrip` and `strrstrip` to ensure compatibility with modern C++ standards.